tests/framework/web/UrlManagerTest.php

```diff
@@ -273,6 +273,34 @@ public function testCreateAbsoluteUrlWithSuffix()
         $this->assertEquals('http://app.example.com/login/', $url);
     }
 
+    public function testCreateAbsoluteUrlWithRelativeHostRule()
+    {
+        $manager = new UrlManager([
+            'baseUrl' => '/',
+            'scriptUrl' => '',
+            'hostInfo' => 'http://example.com',
+            'cache' => null,
+
+            'enablePrettyUrl' => true,
+            'showScriptName' => false,
+            'rules' => [
+                '//example.com/login' => 'site/login',
+            ],
+        ]);
+        $url = $manager->createAbsoluteUrl(['site/login']);
+        $this->assertEquals('http://example.com/login', $url);
+        $url = $manager->createUrl(['site/login']);
+        $this->assertEquals('//example.com/login', $url);
```

Comment:
Yes, but rule, which respond on passed route is contains host. Look at the [another test](https://github.com/yiisoft/yii2/blob/8142de23d432ead57d91530c9324c1f4f03f5928/tests/framework/web/UrlManagerTest.php#L272-L273).